wts/auth_plugins/base.py

```diff
@@ -13,4 +20,15 @@ def find_user(self):
         find user identified in current request
         returns None if no user can be identified
         """
-        return User(userid="test", username="test")
+        # `current_user` validates the token and relies on `OIDC_ISSUER`
+        # to know the issuer
+        default_oauth_client = get_oauth_client(idp="default")
+        flask.current_app.config["OIDC_ISSUER"] = default_oauth_client.metadata[
+            "api_base_url"
+        ].rstrip("/")
+        try:
+            user = current_user
+            return User(userid=user.id, username=user.username)
+        except:
+            #  allow other plugins a chance to authenticate
+            return None
```

Comment:
One thing to note is that for this [plugin to get used](https://github.com/uc-cdis/workspace-token-service/blob/081300e795382b6c9e7d17bb083398489e40a278/wts/api.py#L54) when the `AUTH_PLUGINS` env variable exists, `default` must appear in `AUTH_PLUGINS` (e.g. `AUTH_PLUGINS=default,k8s`).

Right now, cloud automation [hard codes](https://github.com/uc-cdis/cloud-automation/blob/c445e42ad69a8a65eb8846e19cb2d09869f71fd6/kube/services/wts/wts-deploy.yaml#L56) this to `k8s`. Maybe, if necessary, we could configure `AUTH_PLUGINS` per commons using a secret, but I probably think just updating `wts-deploy.yaml` with `AUTH_PLUGINS=default,k8s` will work for now.

Reply:
Can you create an accompanying PR to do that in cloud auto? Will that be a breaking change? e.g. if cloud auto gets updated to that but this new code isn't deployed will it still work okay?

Reply:
I forgot about that. Updating `wts-deploy.yaml` with `AUTH_PLUGINS=default,k8s` would actually be a breaking change because in `master`, the `"test"` `User` is automatically returned so the Kubernetes plugin [wouldn't get a chance to run](https://github.com/uc-cdis/workspace-token-service/blob/081300e795382b6c9e7d17bb083398489e40a278/wts/auth_plugins/__init__.py#L19). Also, if a refresh token did exist in the database for a `"test"` user, anyone could get a `"test"` access token.